test/link/glibc_compat/build.zig

```diff
@@ -17,4 +27,91 @@ pub fn build(b: *std.Build) void {
         _ = exe.getEmittedBin();
         test_step.dependOn(&exe.step);
     }
+
+    // Build & run against a sampling of supported glibc versions
+    for ([_][]const u8{
+        "native-linux-gnu.2.17", // Currently oldest supported, see #17769
+        "native-linux-gnu.2.23",
+        "native-linux-gnu.2.28",
+        "native-linux-gnu.2.33",
+        "native-linux-gnu.2.38",
+        "native-linux-gnu",
+    }) |t| {
+        const target = std.zig.CrossTarget.parse(
+            .{ .arch_os_abi = t },
+        ) catch unreachable;
+
+        const major_ver = target.toTarget().os.version_range.linux.glibc.major;
+        const minor_ver = target.toTarget().os.version_range.linux.glibc.minor;
+
+        std.debug.assert(major_ver == 2); // assuming no v1 or v3 to test
+
+        const exe = b.addExecutable(.{
+            .name = t,
+            .root_source_file = .{ .path = "glibc_runtime_check.zig" },
+            .target = target,
+        });
+        exe.linkLibC();
+
+        // Only try running the test if the host glibc is known to be good enough
+        if (minor_ver <= running_glibc_minor) {
```

Comment:
I wonder if this could be incorporated in our runner much like foreign test runner. What do you think @andrewrk ?

Reply:
It seems complicated to determine the runtime linker's support if there is an emulator in play?  (I guess the test infra could probe for that?)  And I also think these does-runtime-support-newer-glibc checks are probably only needed in this test?  Other tests should just be using the "default" glibc if they're explicitly linking?

That said, I'm willing to try merging this check into the test infra if folks think that is worthwhile.

Reply:
Use the SemanticVersion methods for comparison please. Also, I think you should be able to remove this condition since you have `skip_foreign_checks = true` which is defined to skip the check if the host cannot execute the binary. If the build system needs an improvement to make this work correctly with respect to glibc versions, I'm happy to help with that.

Reply:
I will work on making the `skip_foreign_checks = true` check for the glibc supported versions.  Without this hack here, the CI machines would build binaries that failed to execute.  I can do this as a follow-on to this PR, or will update this PR with the changes if you want to wait for that.

(And I've switched to `SemanticVersion.order` for the version checking.)